mapper/software_management/src/software.rs

```diff
@@ -0,0 +1,133 @@
+use serde::{Deserialize, Serialize};
+
+pub type SoftwareType = String;
+pub type SoftwareName = String;
+pub type SoftwareVersion = String;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct SoftwareModule {
+    #[serde(rename = "type")]
+    pub software_type: SoftwareType,
+
+    pub name: SoftwareName,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub version: Option<SoftwareVersion>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub url: Option<String>,
+}
+
+pub type SoftwareList = Vec<SoftwareModule>;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum SoftwareOperation {
+    // A request for the current software list
+    CurrentSoftwareList,
+
+    // A sequence of updates to be applied
+    SoftwareUpdates { updates: Vec<SoftwareUpdate> },
```

Comment:
I'm assuming that you added this `SoftwareUpdates` variant in addition to the `DesiredSoftwareList` variant to support the clouds that may send just the list of software to be updated rather than sending the entire desired software list. In that case, can we name it something like `IncrementaSoftwareList` or `DeltaSoftwareList` which is more explicit on what the distinction is?